src/fxa-geodb.js

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// these are defaults, can be overloaded by configuring options
+var path = require('path');
+
+var DEFAULTS = require(path.join('..', 'lib', 'defaults'));
+var ERRORS = require(path.join('..', 'lib', 'errors'));
+var maxmind = require('maxmind');
+var Promise = require('bluebird');
+
+module.exports = function (options) {
+  'use strict';
+
+  options = options || {};
+  var dbPath = options.dbPath || DEFAULTS.DEFAULT_DB_PATH;
+
+  var dbLookup, locationData;
+  // we quit if the db did not load for some reason
+  try {
+    dbLookup = maxmind.open(dbPath);
+  } catch (err) {
+    // if it failed with primary database
+    // then quit with error
+    throw ERRORS.UNABLE_TO_FETCH_DATA;
+  }
+
+  return function (ip, userLocale) {
+    return new Promise(function (resolve, reject) {
+      // check if ip is valid
+      if (! maxmind.validate(ip)) {
+        reject({
+          message: ERRORS.IS_INVALID
+        });
+        return;
+      }
+
+      locationData = dbLookup.get(ip);
+
+      if (locationData == null) {
+        reject({
+          message: ERRORS.UNABLE_TO_FETCH_DATA
+        });
+        return;
+      }
+
+      // return an object with city, country, continent,
+      // latitude, and longitude, and timezone
+      var location = new Location(locationData, userLocale);
+
+      resolve(location);
+    });
+  };
+};
+
+function Location(locationData, userLocale) {
```

Comment:
is none of the `userLocale` stuff documented? Should the args to `Location` be `ip, options` because `userLocale` seems really secondary.